src/FakeItEasy.Analyzer/SymbolHelpers.cs

```diff
@@ -18,9 +24,12 @@ internal static IMethodSymbol GetCalledMethodSymbol(InvocationExpressionSyntax c
             var name = (call?.Expression as MemberAccessExpressionSyntax)?.Name
                 ?? call?.Expression as IdentifierNameSyntax;
 
-            return name == null
-                ? null
-                : context.SemanticModel.GetSymbolInfo(name).Symbol as IMethodSymbol;
+            var method =
+                name == null
+                    ? null
+                    : context.SemanticModel.GetSymbolInfo(name).Symbol as IMethodSymbol;
+
+            return method?.ReducedFrom ?? method;

```

Comment:
This is the only actual code change I had to do to make it work in VB.NET. Basically, in Roslyn for VB.NET, a generic extension method is "reduced", i.e. `WhereConfigurationExtensions.Where<T>` becomes `WhereConfigurationExtensions.Where`, as if it were non-generic. It doesn't change anything for C#, where `ReducedFrom` returns null.